File: clientSide/src/app/Models/event.ts

export class Event{

    idEvent? : any;
    name : string;
    description : string;
    startDate : Date;
    endDate : Date;
    nbParticipant : number;
    image? : any;
    price: number;
    category: string;
    idCampPlace?: number;


    constructor(name: string, des: string, startDate: Date, endDate: Date, nbP: number,price: number,cate: string, id?: any,idCampPlace?:number,img?: any) {
        this.idEvent = id;
        this.name = name;
        this.description = des;
        this.startDate = startDate;
        this.endDate = endDate;
        this.nbParticipant = nbP;
        this.price= price;
        this.category=cate;
      }


}

File: apiSide/src/main/java/com/example/tuniscamp/entities/Event.java
package com.example.tuniscamp.entities;
import lombok.*;
import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Setter(AccessLevel.NONE)
    private Integer idEvent;
    private String name;

    @Column(length = 1000)
    private String description;

    private Date startDate ;
    private Date endDate ;
    private  int nbParticipant;
    @Lob
    @Column(columnDefinition = "LONGBLOB")
    private byte[] image;
    private double price;

    // bi
    @ManyToOne
    private User user ;

    @Enumerated(EnumType.STRING)
    private EventCategory category;
    // bi
    @ManyToOne
    private CampPlace campPlace;

    @OneToMany(mappedBy = "event")
    private List<Reservation> reservations;
















}
